packages/flutter/lib/src/painting/oval_border.dart

```diff
@@ -22,6 +25,25 @@ class OvalBorder extends CircleBorder {
   /// Create an oval border.
   const OvalBorder({ super.side, super.ovalness = 1.0});
 
+  @override
+  ShapeBorder scale(double t) => OvalBorder(side: side.scale(t), ovalness: ovalness);
+
+  @override
+  OvalBorder copyWith({ BorderSide? side, double? ovalness }) {
+    return OvalBorder(side: side ?? this.side, ovalness: ovalness ?? this.ovalness);
+  }
+
+  @override
+  ShapeBorder? lerpFrom(ShapeBorder? a, double t) {
+    if (a is CircleBorder) {
+      return OvalBorder(
+        side: BorderSide.lerp(a.side, side, t),
+        ovalness: ui.lerpDouble(a.ovalness, ovalness, t)!,
+      );
+    }
+    return super.lerpFrom(a, t);
+  }
```

Comment:
Not doing this also works

Reply:
Yeah, but the tests fails because those not overridden methods return CircleBorder because you're only overriding toString which has different representation.